File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.kt
class Solution {
    val arr = arrayOf("A", "E", "I", "O", "U")
    val list = mutableListOf<String>()
    
    fun solution(word: String): Int {
        dfs("")
        
        return list.indexOf(word)
    }
    
    fun dfs(input: String) {
        if (input.length > 5) {
            return
        }
        
        list.add(input)
        arr.forEach { dfs(input + it) }
    }
}

File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
words = []
lst = ["A", "E", "I", "O", "U"]

def solution(word):
    dfs("")
    
    return words.index(word)

def dfs(input):
    words.append(input)
    
    if len(input) == 5:
        return
    
    for i in lst:
        dfs(input + i)